[BUG] FW Compile error when enabling verbose spinlock debug

**Describe the bug**
A clear and concise description of what the bug is.

When selecting DEBUG and DEBUG_LOCKS_VERBOSE on the i.MX platform I get the following compile error:

```
[trimmed]/src/include/sof/spinlock.h:87:9: error: 'esai[i].lock' is a pointer; did you mean to use '->'?
   (lock)->user = __LINE__; \
```

(two times in my case, within the macro `spinlock_init`; I think dependent on platform this macro may be called a different number of times but I believe the error is actually not platform-specific)

**To Reproduce**
Steps to reproduce the behavior: (e.g. list commands or actions used to reproduce the bug)
* Generate default configuration: `make imx8_defconfig` (replace with your platform as I don't think this is platform specific)
* Select, in addition to the defconfig, DEBUG, DEBUG_LOCKS ~and DEBUG_LOCKS_VERBOSE~ (also occurs without selecting DEBUG_LOCKS_VERBOSE).
* Attempt build: `make -j8`

**Reproduction Rate**
100% (build error)

**Expected behavior**
Firmware should build successfully

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)
Annoyance

**Environment**
1) Branch name and commit hash of the 2 repositories: sof (firmware/topology) and linux (kernel driver).
    * Kernel: (doesn't matter)
    * SOF: 78b022bf (master)
2) Name of the platform(s) on which the bug is observed.
    * Platform: imx8 (though I believe this is not platform-specific)

--- a/src/arch/xtensa/init.c
+++ b/src/arch/xtensa/init.c
@@ -15,6 +15,7 @@
 #include <sof/init.h>
 #include <sof/lib/cpu.h>
 #include <sof/sof.h>
+#include <sof/spinlock.h>
 #include <config.h>
 #include <xtensa/xtruntime-frames.h>
 #include <xtos-structs.h>

--- a/src/include/sof/spinlock.h
+++ b/src/include/sof/spinlock.h
@@ -15,6 +15,7 @@
 
 #include <arch/spinlock.h>
 #include <sof/drivers/interrupt.h>
+#include <sof/trace/trace.h>
 #include <config.h>
 
 /*
@@ -84,7 +85,7 @@ extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 #define spinlock_init(lock) \
 	do { \
 		arch_spinlock_init(lock); \
-		(lock)->user = __LINE__; \
+		(*lock)->user = __LINE__; \
 	} while (0)
 
 /* panic on deadlock */

--- a/src/include/sof/spinlock.h
+++ b/src/include/sof/spinlock.h
@@ -76,7 +76,7 @@
 #define trace_lock(__e)		trace_error_atomic(TRACE_CLASS_LOCK, __e)
 #define tracev_lock(__e)	tracev_event_atomic(TRACE_CLASS_LOCK, __e)
 #define trace_lock_error(__e)	trace_error_atomic(TRACE_CLASS_LOCK, __e)
-#define trace_lock_value(__e)	_trace_error_atomic(__e)
+#define trace_lock_value(__e)	trace_error_value_atomic(__e)
 
 extern uint32_t lock_dbg_atomic;
 extern uint32_t lock_dbg_user[DBG_LOCK_USERS];

